dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java

```diff
@@ -0,0 +1,353 @@
+    {
+
+        String[] datasetNames = {"~test::index::integer","~test::index::string"};
+        FieldDef[] datasetRecordDefinitions = new FieldDef[2];
+
+        // FieldDef[] fieldDefs = null;
+        // Integer key
+        FieldDef[] fieldDefs = new FieldDef[2];
+        fieldDefs[0] = new FieldDef("key", FieldType.INTEGER, "INTEGER4", 4, true, true, HpccSrcType.LITTLE_ENDIAN, new FieldDef[0]);
+        fieldDefs[1] = new FieldDef("payload", FieldType.STRING, "STRING16", 16, true, false, HpccSrcType.SINGLE_BYTE_CHAR, new FieldDef[0]);
+
+        datasetRecordDefinitions[0]= new FieldDef("RootRecord", FieldType.RECORD, "rec", 4, false, false, HpccSrcType.LITTLE_ENDIAN, fieldDefs);
+
+        // String key
+        fieldDefs = new FieldDef[2];
+        fieldDefs[0] = new FieldDef("key", FieldType.STRING, "STRING4", 4, true, false, HpccSrcType.SINGLE_BYTE_CHAR, new FieldDef[0]);
+        fieldDefs[1] = new FieldDef("payload", FieldType.STRING, "STRING16", 16, true, false, HpccSrcType.SINGLE_BYTE_CHAR, new FieldDef[0]);
+
+        datasetRecordDefinitions[1]= new FieldDef("RootRecord", FieldType.RECORD, "rec", 4, false, false, HpccSrcType.LITTLE_ENDIAN, fieldDefs);
+
+        for (int i = 0; i < datasetNames.length; i++)
+        {
+            //------------------------------------------------------------------------------
+            // Create indexable dataset
+            //------------------------------------------------------------------------------
+
+            ArrayList<HPCCRecord> partitionRangeStart = new ArrayList<HPCCRecord>();
+            ArrayList<HPCCRecord> partitionRangeEnd = new ArrayList<HPCCRecord>();
+
+            String datasetName = datasetNames[i];
+            FieldDef recordDef = datasetRecordDefinitions[i];
+            createIndexableFile(datasetName, recordDef, partitionRangeStart, partitionRangeEnd);
+            
+            //------------------------------------------------------------------------------
+            // Create index
+            //------------------------------------------------------------------------------
+
+            String indexName = datasetName + "::key";
+            String ecl = "rec := " + RecordDefinitionTranslator.toECLRecord(recordDef);
+            ecl += "ds := DATASET('" + datasetName + "', rec, THOR);";
+            ecl += "idx := INDEX(ds, {key}, {payload},'" + indexName + "');";
+            ecl += "BUILDINDEX(idx, OVERWRITE);";
+            
+            WorkunitWrapper wu = new WorkunitWrapper();
+            wu.setECL(ecl);
+            wu.setJobname("IndexCreation" + datasetName);
+            wu.setCluster(thorcluster);
+
+            try
+            {
+                HPCCWsWorkUnitsClient client = wsclient.getWsWorkunitsClient();
+                String result = client.createAndRunWUFromECLAndGetResults(wu);
+            }
+            catch(Exception e)
+            {
+                Assert.fail("Test failed with error: " + e.getMessage());
```

Comment:
It feels like creating the index file should be externalized, and this test voided if creating the index fails.
I think some of the other test use various org.junit.Assume constructs to do this.